applications/segyviewer.py

```diff
@@ -1,410 +1,146 @@
-        self.plotCanvas.set_colormap(str(action))
-
-    def set_vertical_line_indicator(self, line):
-        self.plotCanvas.set_vertical_line_indicator(line)
-
-    def set_horizontal_line_indicator(self, line):
-        self.plotCanvas.set_horizontal_line_indicator(line)
-
-
-class TopMenu(QtGui.QMenuBar):
-    def __init__(self, parent, colormap_monitor):
-        super(QtGui.QMenuBar, self).__init__(parent)
-
-        self.fileMenu = self.addMenu('&File')
-        exitAction = QtGui.QAction('&Exit', self)
-        self.fileMenu.addAction(exitAction)
-
-        self.viewMenu = self.addMenu('&View')
-        self.colormapMenu = self.viewMenu.addMenu("&Colormap")
-
-        self.colormap_monitor = colormap_monitor
-
-        self.colormap_monitor.cmap_changed.connect(self.set_selected_cmap)
+def configure_main_menu(menu, colormap_monitor):
+    menu.fileMenu = menu.addMenu('&File')
+    exitAction = QtGui.QAction('&Exit', menu)
+    menu.fileMenu.addAction(exitAction)
 
-        def colormapChanger(color_map_name):
-            def performColorMapChange():
-                self.colormap_monitor.colormapUpdated(color_map_name)
-            return performColorMapChange
+    menu.viewMenu = menu.addMenu('&View')
+    menu.colormapMenu = menu.viewMenu.addMenu("&Colormap")
 
-        for item in ['seismic', 'spectral', 'RdGy', 'hot', 'jet', 'gray']:
-            action = self.colormapMenu.addAction(item)
-            action.setCheckable(True)
-            action.triggered.connect(colormapChanger(item))
-
-    def set_selected_cmap(self, cmap_name):
-        for item in self.colormapMenu.actions():
+    def set_selected_cmap(cmap_name):
+        for item in menu.colormapMenu.actions():
             item.setChecked(str(item.text()) == cmap_name)
 
+    colormap_monitor.cmap_changed.connect(set_selected_cmap)
 
-class LineSelector(QtGui.QWidget):
-    indexChanged = QtCore.pyqtSignal(int)
-
-    def __init__(self, parent, label, indexes, monitor_func):
-        super(QtGui.QWidget, self).__init__(parent)
-        self.label = label
-        self.indexes = indexes
-        self.monitor_func = monitor_func
-
-        self.layout = QtGui.QHBoxLayout()
-        self.slabel = QtGui.QLabel(self.label)
-        self.sbox = QtGui.QSpinBox(self)
-        self.sbox.setRange(self.indexes[0], self.indexes[-1])
-        self.sbox.valueChanged.connect(self.monitor_func)
-        self.layout.addWidget(self.slabel)
-        self.layout.addWidget(self.sbox)
-        self.setLayout(self.layout)
-
-    def index_changed(self, val):
-        self.indexChanged.emit(val)
+    def colormap_changer(color_map_name):
+        def perform_colormap_change():
+            colormap_monitor.colormap_updated(color_map_name)
 
-    def set_index(self, val):
-        self.sbox.blockSignals(True)
-        self.sbox.setValue(val)
-        self.sbox.blockSignals(False)
+        return perform_colormap_change
 
+    for item in ['seismic', 'spectral', 'RdGy', 'hot', 'jet', 'gray']:
```

Comment:
Perhaps extract `['seismic', 'spectral', 'RdGy', 'hot', 'jet', 'gray']` to a variable at class level?  E.g.
```python
available_colormaps = ['seismic', 'spectral', 'RdGy', 'hot', 'jet', 'gray']
```